src/evm/engine_db/simulation_db.rs

```diff
@@ -94,7 +97,7 @@ pub struct SimulationDB<M: Middleware> {
     /// Client to connect to the RPC
     client: Arc<M>,
     /// Cached data
-    account_storage: RefCell<AccountStorage>,
+    account_storage: Arc<RwLock<AccountStorage>>,
```

Comment:
by doing this I think we are trading off some performance for thread safety and generality 😕 
Do we think the performance decrease will be significant? @zizou0x maybe you have some idea here? (because I have none 😅 )

Reply:
This is the correct way to do it!

Reply:
Agreed with @kayibal, you're right that it adds some overhead but in that particular case the bottleneck will be network by far, and anyone who use this shouldn't expect high performances, so we can afford this IMO.